### DISCUSSION TITLE
[@types/yaireo__tagify] Dependence on types/react

### DISCUSSION THREAD

**Discussion by tobiasdiez** [DefinitelyTyped/DefinitelyTyped#62368] [Issues with a @types package]
Looking at the distributed [package.json](https://unpkg.com/browse/@types/yaireo__tagify@4.16.0/package.json) one sees
```json
"dependencies": {
        "@types/react": "*",
        "@types/yaireo__tagify": "*"
}
```
The self-reference is strange, but not harmful. On the other hand, the dependency on `@types/react` is not needed (the `index.d.ts` file doesn't contain any reference to react types) and may lead to downstream issues such as https://github.com/johnsoncodehk/volar/discussions/592.

Moreover, `dist/react.tagify.d.ts` is not exposed in the `package.json` and thus hard to use (e.g. with `import Tags from '@yaireo/tagify/dist/react.tagify'`). For these reasons, I propose to split the tagify package and extract the react part to a new package.

[a comment by an automated account is not shown]

**Comment by Brakebein** [comment]
I'm not sure, why there is a self-reference. It can probably be removed.

Regarding `dist/react.tagify.d.ts`: it's true that it is hard to use, but this is not our decision, it just reflects the intended use by the orginal package (see https://www.npmjs.com/package/@yaireo/tagify#react). We are not the authors of `@yaireo/tagify`, we only did the type definitions. But we (or at least I) agree with you that such wrappers for third-party libraries like react should be in some separate packages. I don't know, if `@types/react` can be declared as a peer dependency and if this avoids the possible issue you mentioned. @blutorange was responsible for the `react.tagify.d.ts`.

**Reply by tobiasdiez** [reply]
You are of course right, this is only an reflection of the design decision of tagify itself. I've opened https://github.com/yairEO/tagify/issues/1126 to discuss this further.

As a temporary workaround, if declaring the react types as peer dependency works, then this would  be great!

**Reply by blutorange** [reply]
I suppose we should mimic the original package if possible. yairEO/tagify. The [react.tagify.js](https://github.com/yairEO/tagify/blob/master/src/react.tagify.js) file imports from react, but the package.json does not have any reference to react. It should probably contain a peer dependency on react though, so yes, I agree changing `"@types/react"` to a peer dependency is the way to go. Unless it can be extracted in sub packages of course, and that linked issues sounds promising.

As for the self-reference `"@types/yaireo__tagify"`, I'm not sure either. We should just remove it and when that does not cause build issues, that's fine.
